rs/types/types/src/canister_http.rs

```diff
@@ -457,6 +457,10 @@ pub struct CanisterHttpRequest {
     pub id: CanisterHttpRequestId,
     /// The context of the request which captures all the metadata about this request
     pub context: CanisterHttpRequestContext,
+    /// The most up to date api boundary nodes address that should be used as a socks proxy in the case of a request to an IPv4 address.
+    /// The addresses should be sent in the following format: "socks5://[<ip>]:<port>", for example:
+    /// "socks5://[2602:fb2b:110:10:506f:cff:feff:fe69]:1080"
+    pub socks_proxy_addrs: Vec<String>,
```

Comment:
This is a rust type, is it possible to use a type like [Url](https://docs.rs/url/2.5.4/url/) or [SocketAddr](https://doc.rust-lang.org/std/net/enum.SocketAddr.html) ?

Reply:
Not sure they are of much use. We would basically construct and deconstruct them right away. The registry itself doesn't provide such types

Reply:
Why you didn't add the socks_proxy_addrs to the CanisterHttpRequestContext struct instead? It seems CanisterHttpRequestContext is the rust type for the protobuf message